hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java

```diff
@@ -183,18 +176,21 @@ public void run() {
       if (LOG.isInfoEnabled()) LOG.info("Chore: " + getName() + " was stopped");
     } else {
       try {
+        // TODO: Histogram metrics per chore name.
+        // For now, just measure and log if DEBUG level logging is enabled.
+        long start = 0;
+        if (LOG.isDebugEnabled()) {
+          start = System.nanoTime();
+        }
         if (!initialChoreComplete) {
           initialChoreComplete = initialChore();
         } else {
-          timeMeasurement.measure(() -> {
             chore();
-            return null;
-          });
-          if (LOG.isInfoEnabled() && (System.nanoTime() - lastLog > FIVE_MINUTES_IN_NANOS)) {
-            LOG.info("{} average execution time: {} ns.", getName(),
-                (long)(timeMeasurement.getAverageTime()));
-            lastLog = System.nanoTime();
-          }
+        }
+        if (LOG.isDebugEnabled()) {
+          long end = System.nanoTime();
+          LOG.debug("{} execution time: {} ms.", getName(),
```

Comment:
log the start time, too? If the debug was enabled between execution, may get a wrong execution time?

Reply:
Oh, fun edge case @infraio :) How about a check that `start > 0` and only log in that case?

Reply:
Yes we can do that. Will update

Reply:
I was expecting to find something like `if (LOG.isDebugEnabled() && start > 0) {` here. Was this comment addressed elsewhere? Thanks.